typer/main.py

```diff
@@ -873,6 +886,11 @@ def get_click_param(
             annotation=main_type, parameter_info=parameter_info
         )
     convertor = determine_type_convertor(main_type)
+    if lenient_issubclass(main_type, Enum):
+        if parameter_info.enum_by_name:
+            convertor = generate_enum_name_convertor(main_type)
+        else:
+            convertor = generate_enum_convertor(main_type)
```

Comment:
This bit is currently still hacky and not 100% functional. The `determine_type_convertor` is not yet adjusted and always returns `generate_enum_convertor` instead of potentially switching to `generate_enum_name_convertor`. In this code block, that is being overwritten by the additional `if lenient_issubclass(main_type, Enum)` check here, but there's an issue when you'd use an `Enum` inside of a `Tuple`, because `generate_tuple_convertor` will call `determine_type_convertor` directly and thus will not be able to convert the enum correctly.

This can probably be fixed by passing around more parameters, but it might look a bit ugly 🫤  

Let me contribute some of the examples with Tuples+enums I've been testing on, to showcase where the error is (the CI will go red again 😢 )

Reply:
Update: pushed a (somewhat ugly) fix that address this 👇